### PR TITLE
feat(coach): add coding and opt-in general technical skills

### PR DESCRIPTION
**Pull request by EdHasNoLife**
Adds authored Coding guidance and an opt-in **General Technical** format. **None remains the default** and sends the byte-for-byte base coach prompt; the picker offers None, Coding, System Design, Behavioral, and General Technical. Each selected addendum is resolved once at Start. General Technical routes within its own purpose-built prompt using conversation and available screen evidence, without a separate classifier or a default behavior change.

The base prompt owns the action gate, conditional comprehension before strategy, hint length, and healthy-progress silence. Coding supplies format-specific approach/debugging guidance and boundary-test content for an already-warranted post-completion hint; finishing code alone does not trigger speech. General Technical follows the Behavioral skill's completion standard and does not restate a stricter fragment gate. Screen capture remains on demand, and diagrams still require explicit System Design.

Addresses the owner review at https://github.com/JINGBANZ/jarvis/pull/261#issuecomment-5596474618, including removal of the default-routing resolver, format-identity assertions, and corrected canonical wiki documentation. The untracked scenario-matrix claim is removed; no model scenario results are claimed for this revision.

Validation:
- Restored the original None/base-prompt test unchanged; it failed before the default fix and passes afterward.
- `swift build && ./scripts/run-tests.sh`: build passed; **974 tests in 114 suites passed**.
- Resource and composed-prompt tests identify each selected format by its heading. Preference tests cover all explicit formats, and diagram tests include General Technical and None as disallowed formats.
- Signed-app live General Technical practice remains a manual smoke check before release; it is recorded in wiki/status.md.

### CODE REVIEW

File: wiki/settings-window.md [hunk 1]
```diff
@@ -214,8 +214,12 @@ per-thread `model_reasoning_effort`; both CLI scales start at `low`, so None cla
...
 
 **Interview format.** A second Coaching-card picker (`InterviewFormat`: **None**, plus one entry per
-format that actually has content — System Design today) supplies additional coaching-prompt
-vocabulary for the selected format. **None** persists as no selection and resolves to no addendum at
+format that actually has content — Coding and System Design today) supplies additional
+coaching-prompt vocabulary for the selected format. Coding teaches the coach to distinguish prompt
+comprehension, approach selection, local implementation blocks, visible defects, implementation
+completion, and healthy progress from the transcript and current screen. Its spoken hints stay to
+three short lines, and a completed implementation prompts focused boundary tests when testing has
+not already been discussed. **None** persists as no selection and resolves to no addendum at
```
[the remark this answers is not in the document]
Comment (EdHasNoLife) [reply wiki/settings-window.md]: Consolidated the Settings page to selection/session semantics and linked the canonical policy in architecture.md in d4bd7fe.

File: wiki/status.md [hunk 2]
```diff
@@ -10,6 +10,11 @@
 
 **General technical-interview coaching, audio reliability, and local CLI brain providers are
 implemented.** The coach covers behavioral, system-design, and coding questions. A direct request
+in a session explicitly configured as Coding uses a three-line, one-way coding skill that explains
+an untouched prompt before proposing an approach, preserves valid work in progress, pinpoints clear
+visible defects, prompts focused tests after an apparently complete implementation, and stays silent
+while the candidate is progressing. The selected brain infers the candidate's need from the existing
+transcript and screen; there is no added runtime classifier or candidate dialog. A direct request
```
[the remark this answers is not in the document]
Comment (EdHasNoLife) [reply wiki/status.md]: Replaced the duplicated status prose with the canonical architecture.md link, including the Built file pointer, in d4bd7fe.

File: Tests/JarvisCoreTests/Config/InterviewFormatTests.swift [hunk 3]
```diff
@@ -8,21 +8,27 @@ import Testing
...
-        #expect(InterviewFormat.coding.promptAddendum.isEmpty)
+    /// A missing skill remains a normal empty state, while each authored format becomes available
+    /// through the same resource lookup the Settings picker consumes.
+    @Test func authoredFormatsHaveContent() {
+        #expect(!InterviewFormat.coding.promptAddendum.isEmpty)
         #expect(InterviewFormat.behavioral.promptAddendum.isEmpty)
         #expect(!InterviewFormat.systemDesign.promptAddendum.isEmpty)
-        #expect(InterviewFormat.systemDesign.promptAddendum.contains("functional requirements"))
-        #expect(InterviewFormat.systemDesign.promptAddendum.contains("API"))
     }
```
[the remark this answers is not in the document]
Comment (EdHasNoLife) [reply Tests/JarvisCoreTests/Config/InterviewFormatTests.swift:23]: Pinned system-design resource identity with an API marker and Coding identity with its skill heading in d4bd7fe. The composed Coding system-prompt test now checks that same heading while retaining the base-prefix assertion.

File: wiki/settings-window.md [hunk 4]
```diff
@@ -213,12 +213,16 @@ provider is active; its default lives with the others in
...
+**Interview format.** A second Coaching-card picker defaults to **Automatic**, followed by one
+explicit override per format with authored content — Coding and System Design today. Automatic
+selects appropriate coding, system-design, behavioral, or neutral coaching from the newest speech
+and current screen on every response; it needs no question-boundary signal and does not carry the
+previous response's inferred format forward. Ambiguous evidence stays on the base policy. Coding
+distinguishes comprehension, approach selection, local implementation blocks, visible defects,
+completion tests, and healthy progress; System Design tracks the current design stage. The resolved
+prompt is fixed at Start so CLI and OpenAI providers share identical instructions, but Automatic's
+model-level choice changes with current evidence inside that prompt. Selecting Coding or System
+Design remains an optional next-Start override. See
```
[the remark this answers is not in the document]
Comment (EdHasNoLife) [reply wiki/settings-window.md]: Removed the per-state enumeration from Settings and linked architecture.md as the canonical per-format policy in d4bd7fe.

File: Sources/JarvisCore/Resources/Skills/automatic.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+For a behavioral task, organize the answer as STAR: Situation, Task, Action, Result. A complete new
+interviewer question is a useful coaching moment; give a compact answer direction before the
+candidate starts. When prep search is available, use candidate-owned events as real stories and
+company values, leadership principles, role expectations, or behavioral requirements as answer
+criteria. Criteria alone are not a personal story: if no real story is available, clearly label any
+constructed mini-story “Illustrative example.” Turn partial candidate facts into a coherent framing
+without inventing personal details, outcomes, or metrics. As the candidate answers, speak only for
+one material gap such as missing ownership, vague action, absent result, weak evidence, or criteria
+misalignment. Once all four parts, a specific action, and a concrete result are present, stay silent
+rather than request optional polish.
```
[the remark this answers is not in the document]
Comment (EdHasNoLife) [reply Sources/JarvisCore/Resources/Skills/automatic.md]: Fixed in d898bcb. Automatic mode now stays silent only when no material gap remains, explicitly including criteria misalignment.

File: wiki/settings-window.md [hunk 6]
```diff
@@ -213,12 +213,14 @@ provider is active; its default lives with the others in
...
-for the selected format. **None** persists as no selection and resolves to no addendum at
-all, so a user who never opens this setting sees no behavior change; it is not a guess assembled
-from whatever formats happen to have content. Fixed for the whole session, like the transcription
-language/model choice: it applies on the next Start, never reclassified mid-conversation. See
+**Interview format.** A second Coaching-card picker defaults to **Automatic**, followed by one
+explicit override per format with authored content — Coding, Behavioral, and System Design today. Automatic
+selects appropriate coding, system-design, behavioral, or neutral coaching from the newest speech
+and current screen on every response; it needs no question-boundary signal and does not carry the
+previous response's inferred format forward. Ambiguous evidence stays on the base policy. Coding
+Behavioral, and System Design remain optional next-Start overrides. The resolved prompt is fixed at Start so
```
[the remark this answers is not in the document]
Comment (EdHasNoLife) [reply wiki/settings-window.md]: Fixed in d898bcb by adding the missing comma.

**Comment by JINGBANZ** [comment]
Thanks Ed. The coding skill and the single purpose-built routing file are a clear step up from the concatenation in #252. Two structural changes, then prompt cleanup.

**1. Keep None as the default. Ship the routing skill as a fifth format called "General Technical".**
`nil` should keep meaning "no addendum, base prompt only", so a user who never opens Settings sees no change (same reasoning as the #252 review). Add a `generalTechnical` case to `InterviewFormat` with its own `.md`, next to Coding, Behavioral, and System Design. That removes `resolvedPromptAddendum(for:)`, restores the byte-for-byte None test unchanged, and keeps the wiki rationale intact. Please also retitle or split the PR: the squash subject will say "add coding interview skill" while the main effect is a default change.

**2. Remove the rules that override the base action policy.**
- `automatic.md` L18-20 and `coding.md` L25-26 say "call speak" when the implementation looks complete. A finished implementation is not a stuck moment, and base steps 4 to 6 already handle the real cases: a stall after finishing, an interviewer asking about tests, or the hint key. Keep the hint shape (spend a post-completion hint on boundary tests, no praise or complexity) and drop the trigger.
- Comprehension-first is unconditional in both skills but conditional on "long or dense" in base Tip style (L58-61). One rule, stated once.
- `automatic.md` L40-42 "must call stay_silent" for fragments is stricter than the base gate and would silence a cut-off "Jarvis, ...". Drop the restatement.

**3. Cut generic rules from the skill files.**
The three-line cap, stay-silent-on-progress, and meaning-then-strategy already live in base and the tool descriptions. The coding policy now exists in three copies with different conditions. Skill files should carry only format-specific behavior. Also `automatic.md` L37-38 flips `behavioral.md` L31-33 on when to go silent; reconcile them.

**4. Tests and wiki.**
- "functional requirements" and "STAR" now appear in two skill files, so `CoachDriverInterviewFormatTests` L37 cannot tell which one loaded. Assert the `# Interview format: ...` heading instead. `CoachDriverDiagramTests` L19 still uses `?? ""`.
- Wiki says the routing skill uses "the current screen on every response"; capture is on demand. Note that diagram hints still require explicit System Design (`CoachAttemptRunner` L289, L415). The "scenario matrix" cited in `status.md` is not in the repo; drop the reference or paste per-scenario pass counts in the PR body as #262 did.

File: wiki/status.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (EdHasNoLife) [reply wiki/status.md]: Verified on the current #261 head (4cebe65): status contains the canonical architecture cross-reference and the Built source pointer, without repeating the coding-state enumeration. The earlier fix is retained; resolving this thread.

File: Tests/JarvisCoreTests/Config/InterviewFormatTests.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (EdHasNoLife) [reply Tests/JarvisCoreTests/Config/InterviewFormatTests.swift:23]: Verified on 4cebe65: authoredFormatsHaveExpectedContent checks both functional requirements and API for System Design, plus the Coding heading. The focused resource/composed-prompt run completed with 11 tests in 3 suites passing.

File: wiki/settings-window.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (EdHasNoLife) [reply wiki/settings-window.md]: Verified on 4cebe65: Settings links to architecture.md#models-and-apis for per-format coaching behavior, and status uses the same canonical reference. The duplicated coding sub-state enumeration is gone; resolving this thread.

**Comment by EdHasNoLife** [comment]
Addressed the specific [owner review](https://github.com/JINGBANZ/jarvis/pull/261#issuecomment-5596474618) in **8c380da**. My earlier status checked inline review threads and missed this top-level comment.

1. **Default and format:** None again means nil/no addendum/base prompt only. General Technical is a separate opt-in enum case and `general-technical.md`, the fifth picker option including None. Removed `resolvedPromptAddendum(for:)` and retitled/reworked the PR description.
2. **Action policy:** Removed completion-triggered `speak`, unconditional comprehension-first, and the routing skill's stricter fragment-silence rule. Base action and Tip style policy now own those decisions; post-completion guidance only specifies the content of an already-warranted hint.
3. **Prompt cleanup:** Removed generic hint caps and progress/meaning-then-strategy restatements from the coding/routing skills. General Technical uses the Behavioral skill's concrete, candidate-owned, complete, aligned completion standard without requesting extra refinement.
4. **Tests and wiki:** Restored the original byte-for-byte None test unchanged and observed it fail before the fix. Resource/composed-prompt assertions use format headings. Diagram tests include General Technical and None as disallowed formats; `format?.promptAddendum ?? ""` now matches the restored production None behavior. Wiki documents on-demand capture and explicit-System-Design-only diagrams. Removed the untracked scenario-matrix claim from the wiki and PR body.

Validation: `swift build && ./scripts/run-tests.sh` completed successfully — **974 tests in 114 suites passed**. Independent review found no remaining omissions against this comment. A live signed-app practice smoke remains before release; no model scenario pass counts are claimed for this revised prompt.

**Review by JINGBANZ — APPROVED** [review]
